### PR TITLE
Billing for render

### PR DESCRIPTION
**Pull request by abdullah248**
## Description
Sets render properties that will be used for billing

## Related issues
Addresses [[AB#102960](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/102960)].

## Testing
Added unit tests

### CODE REVIEW

File: src/Microsoft.Health.Dicom.Api.UnitTests/Features/Context/DefaultDicomRequestContext.cs [hunk 1]
```diff
@@ -26,6 +26,10 @@ public class DefaultDicomRequestContext : IDicomRequestContext
 
     public long BytesTranscoded { get; set; }
 
+    public bool IsRenderedRequested { get; set; }
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Api.UnitTests/Features/Context/DefaultDicomRequestContext.cs:29]: You need to add the same in workspaceplatform too

Comment (abdullah248) [reply src/Microsoft.Health.Dicom.Api.UnitTests/Features/Context/DefaultDicomRequestContext.cs:29]: Yep, going to cut a release and pull this into workspace platform once this pr gets checked in

File: src/Microsoft.Health.Dicom.Core/Features/Retrieve/RetrieveRenderedService.cs [hunk 2]
```diff
@@ -96,6 +97,9 @@ public async Task<RetrieveRenderedResponse> RetrieveRenderedImageAsync(RetrieveR
             sw.Stop();
             _logger.LogInformation("Render from dicom to {OutputContentType}, uncompressed file size was {UncompressedFrameSize}, output frame size is {OutputFrameSize} and took {ElapsedMilliseconds} ms", outputContentType, stream.Length, resultStream.Length, sw.ElapsedMilliseconds);
 
+            _dicomRequestContextAccessor.RequestContext.IsRenderedRequested = true;
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Core/Features/Retrieve/RetrieveRenderedService.cs:100]: Not sure why needed this boolean, if the length is greater than 0, we can bill them. Whats your thought?

File: src/Microsoft.Health.Dicom.Core.UnitTests/Features/Retrieve/RetrieveRenderedServiceTests.cs [hunk 3]
```diff
@@ -314,6 +322,8 @@ public async Task GivenStoredInstancesWithFramesPNG_WhenRetrieveRenderedForFrame
         resultStream.Position = 0;
         AssertStreamsEqual(resultStream, response.ResponseStream);
         Assert.Equal("image/png", response.ContentType);
+        Assert.True(_dicomRequestContextAccessor.RequestContext.IsRenderedRequested);
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Core.UnitTests/Features/Retrieve/RetrieveRenderedServiceTests.cs:325]: Could you also assert that PartCount == 1

File: src/Microsoft.Health.Dicom.Core/Features/Retrieve/RetrieveRenderedService.cs (hunk 2 above)
Comment (abdullah248) [reply src/Microsoft.Health.Dicom.Core/Features/Retrieve/RetrieveRenderedService.cs:100]: yeah thats true, was just following the same logic as we have for transcode - but I dont see any real need for it in workspace-platform side either

File: src/Microsoft.Health.Dicom.Core.UnitTests/Features/Retrieve/RetrieveRenderedServiceTests.cs (hunk 3 above)
Comment (abdullah248) [reply src/Microsoft.Health.Dicom.Core.UnitTests/Features/Retrieve/RetrieveRenderedServiceTests.cs]: added

**Review by bcarthic — APPROVED** [review]

**Pull request merged**